<h2>Why is this an issue?</h2>
<p>Each element should be on a line to itself.</p>
<h3>Noncompliant code example</h3>
<pre>
&lt;parent&gt;&lt;child /&gt;&lt;/parent&gt;  &lt;!-- Noncompliant --&gt;
</pre>
<h3>Compliant solution</h3>
<pre>
&lt;parent&gt;
  &lt;child /&gt;
&lt;/parent&gt;
</pre>

